### PR TITLE
Browser Home Page accepts wider variety of URLs

### LINKED ISSUE #331: Default Search Engine Field Too strict
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [!] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/meilix

**Actual Behaviour**

When you enter your webpage, people won't really suppose they will have to type the https:// tag in. Maybe remove some restrictions, or maybe add the html:// tag in front of the website

**Expected Behaviour**

If I type www.google.com it should accept it

**Steps to reproduce it**

Just type www.google.com into that field and you will see the issue

**Related code:**

<input pattern="https?://.+" class="form-control" placeholder="Enter link" name="search_engine" title="only url is allowed" type="url" autocomplete="off">

**Other information:**

I guess might want to post another suggestion here: you should turn on autocomplete and it should be something like if it is not completed then fill 'www.google.com' in

### PR DESCRIPTION
**Pull request by PulkitMishra**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (provide a screenshot or link for test) 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Solves issue #331 
Browser Home page input field now accepts urls with or without https
As @imahumanrcw pointed out and @meets2tarun @sarnava1 seemed to agree that people normally enter 'www.google.com' instead of 'https://www.google.com/' (just an example) so the website must take that into account people entering the former type of url also.

#### Changes proposed in this pull request:

- Updated Regular Expression Value in the pattern attribute.

Fixes #331 

[Preview](https://shrouded-atoll-72279.herokuapp.com/)

### CODE REVIEW

**Comment by PulkitMishra** [comment]
@pranav1698 I had provided the wrong link in preview initially, apologies for the same . Please check now.
Also by all url inputs do you mean the one for default search engine?
If that is the  case consider it done.

**Comment by PulkitMishra** [comment]
@pranav1698 seems to work. Please check.
Although I don't quite understand, pattern is supposed to override input type. Nonetheless, your suggestion works as required.

File: templates/index.html [hunk 1]
```diff
@@ -29,17 +29,17 @@
...
           </div>
           <br>
           <div class="form-group mouse-over-text">
             <label class="heading" for="search_engine">Default Search Engine</label>
             <div class="input-group margin-bottom-sm">
               <span class="input-group-addon"><i class="fa fa-search fa-fw"></i></span>
-              <input pattern="https?://.+" class="form-control" placeholder="Enter link" name="search_engine" data-toggle="tooltip" title="only url is allowed"
-                 type="url"></div>
+              <input pattern="^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$" class="form-control" placeholder="Enter link" name="search_engine" data-toggle="tooltip" title="only url is allowed"
+                 type="text"></div>
```
Comment (r0mflip) [inline templates/index.html]: Why is the type set to text, `type="url"` is the one that should be

File: templates/index.html [hunk 2]
```diff
@@ -29,17 +29,17 @@
             <label class="heading" for="event_url">Browser Home Page </label>
             <div class="input-group margin-bottom-sm">
               <span class="input-group-addon"><i class="fa fa-window-maximize fa-fw"></i></span>
-              <input pattern="https?://.+" class="form-control" placeholder="Enter link" name="event_url" data-toggle="tooltip" title="only url is allowed"
-                 type="url"></div>
+              <input pattern="^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$" class="form-control" placeholder="Enter link" name="event_url" data-toggle="tooltip" title="only url is allowed"
```
Comment (r0mflip) [inline templates/index.html]: Why is the type set to text, `type="url"` is the one that should be

**Pull request merged**